python/rascaline/rascaline/utils/power_spectrum/calculator.py

```diff
@@ -287,23 +312,28 @@ def _positions_gradients(new_block, block_1, block_2, factor):
             grad2_sample_idxs,
         ) = gradient_1.samples.union_and_mapping(gradient_2.samples)
 
-        gradient_values = np.zeros(
-            [gradients_samples.values.shape[0], 3, len(new_block.properties)]
+        gradient_values = _dispatch.zeros_like(
+            array=gradient_1.values,
+            shape=[gradients_samples.values.shape[0], 3, len(new_block.properties)],
         )
 
         # this is equivalent to an einsum with: ixma, imb -> ixab
-        gradient_1_values = factor * np.matmul(
+        block_2_values = block_2.values[gradient_1.samples.column("sample")]
```

Comment:
you might have to do the same as in https://github.com/lab-cosmo/equistore/pull/332 with `_dispatch.to_index_array` here. This is only an issue on older torch version

Reply:
Ohh yes thanks!